update the limitations of multi-schema change (#9852)
pingcap/docs: sql-statements/sql-statement-alter-table.md

The following major restrictions apply to `ALTER TABLE` in TiDB:

- When altering multiple schema objects in a single `ALTER TABLE` statement:

>     - Changing the same column or index in a single statement is not supported.

Reviewer: ```suggestion
    - Modifying the same object in multiple changes is not supported.
```

>     - TiDB validates statements according to the table schema **before execution**. For example, an error returns when executing `ALTER TABLE ADD INDEX i(b), DROP INDEX i;` due to the index `i` is not existed in the table.

Reviewer: ```suggestion
    - TiDB validates statements according to the table schema **before execution**. For example, an error returns when `ALTER TABLE ADD INDEX i(b), DROP INDEX i;` is executed because the index `i` does not exist in the table.
```

>     - The execution order of TiDB is to execute changes one by one from left to right, which is incompatible with MySQL in some cases.

Reviewer: ```suggestion
    - For an `ALTER TABLE` statement, the order of execution in TiDB is one change after another from left to right, which is incompatible with MySQL in some cases.
```
